### PR TITLE
Remove helm NicClusterPolicy install from docs

### PR DESCRIPTION
**Pull request by killianmuldoon**
Remove the instructions for installing the NicClusterPolicy through helm from the documentation and replace it with a two step process that involves applying the NicClusterPolicy directly using `kubectl`.

This feature [is deprecated and is going to be removed from the Network Operator helm chart in a future release](https://github.com/mellanox/network-operator/blob/d55b311ece00be89b545f62ed71dad12bc4b7087/deployment/network-operator/templates/NOTES.txt#L6-L12)

### CODE REVIEW

File: docs/getting-started-kubernetes.rst [hunk 1]
```diff
@@ -156,142 +156,252 @@ Below are deployment examples, which the ``values.yaml`` file provided to the He
...
+.. code-block:: yaml
+    apiVersion: mellanox.com/v1alpha1
+    kind: NicClusterPolicy
+    metadata:
+      name: nic-cluster-policy
+    spec:
+      ofedDriver:
+        image: doca-driver
+        repository: nvcr.io/nvidia/mellanox
+        version: 24.04-0.6.6.0-0
```
Comment (killianmuldoon) [inline docs/getting-started-kubernetes.rst:184]: I have to update this by templating the version correctly.

File: docs/getting-started-kubernetes.rst [hunk 2]
```diff
@@ -156,142 +156,256 @@ Below are deployment examples, which the ``values.yaml`` file provided to the He
...
+First install the Network Operator with NFD enabled:
 
-``values.yaml`` configuration file for such a deployment:
+``values.yaml``:
 
 .. code-block:: yaml
 
     nfd:
       enabled: true
-    sriovNetworkOperator:
```
Comment (rollandf) [inline docs/getting-started-kubernetes.rst:167]: Is the default `false` for sriovNetworkOperator?

Comment (killianmuldoon) [reply docs/getting-started-kubernetes.rst:167]: Yes - https://github.com/mellanox/network-operator/blob/56e4a72472f7746cece5513468e6386605b4d729/deployment/network-operator/values.yaml#L25-L26

File: docs/getting-started-kubernetes.rst [hunk 3]
```diff
@@ -156,142 +156,256 @@ Below are deployment examples, which the ``values.yaml`` file provided to the He
...
+Once the Network Operator is installed create a NicClusterPolicy with
+* DOCA driver
+* RDMA Shared device plugin configured to a netdev with name ens1f0.
+
 
-    sriovDevicePlugin:
-      deploy: false
+Note: You may need to change the interface names in the NicClusterPolicy to those used by your target nodes.
+
+.. parsed-literal::
```
Comment (tobiasgiese) [inline docs/getting-started-kubernetes.rst:175]:
```suggestion
.. code-block:: yaml
```

shouldn't this be rather a `yaml` code block?

Comment (killianmuldoon) [reply docs/getting-started-kubernetes.rst:175]: Unfortunately no - as we need to substitute the version numbers here. That functionality doesn't work with code-block. So here we're trading off the variable substitution for syntax highlighting.

Comment (killianmuldoon) [reply docs/getting-started-kubernetes.rst:175]: https://github.com/Mellanox/network-operator-docs/issues/28

Comment (tobiasgiese) [reply docs/getting-started-kubernetes.rst:175]: Okay, wasn't aware of that. Thanks

**Review by rollandf — APPROVED** [review]

**Comment by krembu** [comment]
@rollandf @adrianchiris please add me as reviewer for upcoming doc changes.

here few comments we should get address in upcoming changed:

- Adjust doc to NFD install externally to Network Operator.
- We set NV IPAM as our default IPAM but documentation still vastly refer to whereabouts

**Comment by adrianchiris** [comment]
@noama-nv please star the project to get notifications. you can then get notified on changes.

**Comment by noama-nv** [comment]
@adrianchiris  i am, but in this case it was reviewed and merge at the speed of light :)
think you should include me in default reviewers

**Pull request merged**